src/Core/SqlManager/Form/SqlRequestFormFactory.php

```diff
@@ -0,0 +1,89 @@
+<?php
+/**
+ * 2007-2018 PrestaShop.
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\SqlManager\Form;
+
+use PrestaShop\PrestaShop\Core\Form\IdentifiableObjectFormDataProviderInterface;
+use PrestaShop\PrestaShop\Core\Form\IdentifiableObjectFormFactoryInterface;
+use PrestaShop\PrestaShop\Core\Hook\HookDispatcherInterface;
+use PrestaShopBundle\Form\Admin\Configure\AdvancedParameters\RequestSql\SqlRequestType;
+use Symfony\Component\Form\FormFactoryInterface;
+
+/**
+ * Class SqlRequestFormFactory creates form used in Back Office for adding/editing SqlRequests.
+ */
+final class SqlRequestFormFactory implements IdentifiableObjectFormFactoryInterface
+{
+    /**
+     * @var FormFactoryInterface
+     */
+    private $formFactory;
+
+    /**
+     * @var HookDispatcherInterface
+     */
+    private $hookDispatcher;
+
+    /**
+     * @var IdentifiableObjectFormDataProviderInterface
+     */
+    private $sqlRequestFormDataProvider;
+
+    /**
+     * @param FormFactoryInterface $formFactory
+     * @param HookDispatcherInterface $hookDispatcher
+     * @param IdentifiableObjectFormDataProviderInterface $sqlRequestFormDataProvider
+     */
+    public function __construct(
+        FormFactoryInterface $formFactory,
+        HookDispatcherInterface $hookDispatcher,
+        IdentifiableObjectFormDataProviderInterface $sqlRequestFormDataProvider
+    ) {
+        $this->formFactory = $formFactory;
+        $this->hookDispatcher = $hookDispatcher;
+        $this->sqlRequestFormDataProvider = $sqlRequestFormDataProvider;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function create(array $data, $id = null)
```

Comment:
what is the use of `$data` ? if the `FormDataProvider` manages the data through `getData($id)` maybe this interface should only be `public function create($id = null)`